### PR TITLE
(MODULES-2658) Detect DSC pending reboot state

### PR DESCRIPTION
**Pull request by Iristyle**

### CODE REVIEW

**Comment by jpogran** [comment]
LGTM

File: spec/unit/provider/reboot/windows_spec.rb [hunk 1]
```diff
@@ -343,6 +343,53 @@ def expects_registry_value_not_found(path, name)
...
+      let(:lcm)             { stub('lcm') }
+      let(:ole_config)      { stub('ole_config') }
+      let(:dsc_meta_config) { stub('dsc_meta_config') }
+
+      describe 'when DSC is available on the system' do
+        before :each do
+          WIN32OLE.expects(:connect).with(root).returns(dsc)
+          dsc.expects(:Get).with('MSFT_DSCLocalConfigurationManager').returns(lcm)
+          lcm.expects(:ExecMethod_).with('GetMetaConfiguration').returns(ole_config)
+          ole_config.expects(:MetaConfiguration).returns(dsc_meta_config)
```
Comment (ferventcoder) [inline spec/unit/provider/reboot/windows_spec.rb]: Stub stub stub it out.

File: spec/unit/provider/reboot/windows_spec.rb [hunk 2]
```diff
@@ -343,6 +343,53 @@ def expects_registry_value_not_found(path, name)
...
+        end
+
+        it 'does not reboot when MSFT_DSCLocalConfigurationManager class is inaccessible' do
+          dsc = stub('dsc')
+          WIN32OLE.expects(:connect).with(root).returns(dsc)
+          dsc.expects(:Get).with('MSFT_DSCLocalConfigurationManager').raises
+
+          provider.should_not be_pending_dsc_reboot
+        end
+      end
```
Comment (ferventcoder) [inline spec/unit/provider/reboot/windows_spec.rb]: Isn't there one additional scenario here? Where DSC is accessible, but LCMState doesn't have any information available for reboot (e.g. WMF 4)?

File: spec/unit/provider/reboot/windows_spec.rb (hunk 1 above)
Comment (Iristyle) [inline spec/unit/provider/reboot/windows_spec.rb]: Sadly, not much choice here.

File: spec/unit/provider/reboot/windows_spec.rb (hunk 2 above)
Comment (Iristyle) [inline spec/unit/provider/reboot/windows_spec.rb]: Technically you're probably right... but these tests all do basically the same thing.

Comment (ferventcoder) [inline spec/unit/provider/reboot/windows_spec.rb]: No worries. :)

**Pull request merged**